### PR TITLE
Initial conversion of parsing to lambdas

### PR DESCRIPTION
**Pull request by leb2**
These changes are moved over from my other PR.
Tested the test cases that I wrote.

### CODE REVIEW

File: tests/test_formula_parser.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        func = p.parse('100 > 10.0')['result']
+        result = func({})
+        self.assertEqual(result, True)
+
+        p = Parser(debug=True)
+        func = p.parse('A > 10.0')['result']
+        result = func({'A': 100})
+        self.assertEqual(result, True)
+
+    def test_variable_name(self):
```
Comment (leb2) [inline tests/test_formula_parser.py:82]: I ran this issue while working on the backend, but this test case doesn't pass yet. I tried again in the original library and it worked, so I'm not sure it is the fault of the lexer. I'm still looking into this one.

Comment (ucwillg) [reply tests/test_formula_parser.py:82]: what does it return?

File: tests/test_formula_parser.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import unittest
+
+from hotxlfp import Parser
+
+
+class TestFormulaParser(unittest.TestCase):
```
Comment (ucwillg) [inline tests/test_formula_parser.py:6]: date calculations would be a nice thing to test next

File: hotxlfp/grammarparser/parser.py [hunk 3]
```diff
@@ -56,7 +57,7 @@ def parse(self, input):
     def run(self):
         while 1:
             try:
-                s = raw_input('=')
+                s = input('=')
```
Comment (ucwillg) [inline hotxlfp/grammarparser/parser.py:61]: what's the reason for this change?

separetely, what does this method do? is this a REPL mode or something?

Comment (leb2) [reply hotxlfp/grammarparser/parser.py:61]: I'm not sure, but I was getting lint errors before because raw_input doesn't not exist. I think it is a python2 vs python3 thing.

File: tests/test_formula_parser.py (hunk 1 above)
Comment (leb2) [reply tests/test_formula_parser.py:82]: It just returns None, when there are numbers in the variable name. This is a problem when converting variables to x1, x2, etc.

Comment (leb2) [reply tests/test_formula_parser.py:82]: Managed to find out the reason for this. Turns out if there was a number it was being parsed as a cell instead of a variable.

**Review by ucwillg — APPROVED** [review]
